app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt

```diff
@@ -188,27 +191,72 @@ class DraftController @Inject constructor(
             return@let document.outerHtml()
         } ?: ""
 
-        val ccList = if (message.cc.isNotEmpty()) {
-            "<div>$ccTitle ${message.cc.joinToString { it.quotedDisplay() }}<br></div>"
-        } else {
-            ""
+        val previousFullBody = computePreviousFullBody(previousBody, message)
+
+        return assembleForwardHtmlHeader(
+            messageForwardHeader,
+            fromTitle,
+            message,
+            dateTitle,
+            subjectTitle,
+            toTitle,
+            ccTitle,
+            previousFullBody
+        )
+    }
+
+    private fun assembleForwardHtmlHeader(
+        messageForwardHeader: String,
+        fromTitle: String,
+        message: Message,
+        dateTitle: String,
+        subjectTitle: String,
+        toTitle: String,
+        ccTitle: String,
+        previousFullBody: String
+    ): String {
+        val forwardRoot = "<div class=\"${MessageBodyUtils.INFOMANIAK_FORWARD_QUOTE_HTML_CLASS_NAME}\" />"
+        return newDocumentWithParentElement(forwardRoot).apply {
+            addAndEscapeTextLine("---------- $messageForwardHeader ---------")
+            addAndEscapeTextLine("$fromTitle ${message.fromName()}")
+            addAndEscapeTextLine("$dateTitle ${message.date.toDate()}")
+            addAndEscapeTextLine("$subjectTitle ${message.subject}")
+            addAndEscapeRecipientLine(toTitle, message.to)
+            addAndEscapeRecipientLine(ccTitle, message.cc)
+            addAndEscapeTextLine("")
+            addAndEscapeTextLine("")
+
+            addAlreadyEscapedBody(previousFullBody)
+        }.outerHtml()
+    }
+
+    private fun newDocumentWithParentElement(parentElementHtml: String): Element {
+        val doc = Jsoup.parseBodyFragment(parentElementHtml)
+        return doc.body().firstElementChild()!!
+    }
+
+    private fun Element.addAndEscapeTextLine(content: String, endWithBr: Boolean = true) {
+        appendElement("div").apply {
+            text(content)
+            if (endWithBr) appendElement("br")
         }
+    }
 
-        val previousFullBody = computePreviousFullBody(previousBody, message)
+    private fun Element.addAndEscapeRecipientLine(prefix: String, recipientList: List<Recipient>) {
+        formatRecipientList(recipientList)?.let { recipients -> addAndEscapeTextLine("$prefix $recipients") }
+    }
+
+    private fun formatRecipientList(recipientList: List<Recipient>): String? {
+        return if (recipientList.isNotEmpty()) recipientList.joinToString { it.quotedDisplay() } else null
+    }
+
+    private fun Element.addAlreadyEscapedBody(previousFullBody: String) {
+        append(previousFullBody)
+    }
```

Comment:
Is this function really useful ? Doesn't seems like it
If any, a commentary would be enough to tell it was already escaped

Reply:
This is mainly because we need to make sure both places where this is called always do the same things and they must never diverge. It just so happens that the thing that needs to be done is quiet simple, at least for now. 

Also yes the function name gives a bit more insight, but as you said, it could be conveyed using a comments as well